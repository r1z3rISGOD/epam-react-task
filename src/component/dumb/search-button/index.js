import {SearchButton} from "./search-button"
export {SearchButton}
