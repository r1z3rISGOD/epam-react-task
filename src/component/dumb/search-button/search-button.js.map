{"version":3,"file":"search-button.js","sourceRoot":"","sources":["search-button.tsx"],"names":[],"mappings":";;;AAAA,6BAA8B;AAC9B,gCAA6B;AAStB,IAAM,YAAY,GAAiC,UAAC,EAAmD;QAAjD,OAAO,aAAA,EAAE,OAAO,aAAA,EAAE,WAAW,iBAAA,EAAE,gBAAgB,sBAAA;IAC1G,OAAO,CACD,gCAAQ,OAAO,EAAE;YACf,IAAI,OAAO,KAAK,EAAE,EAAE;gBAClB,gBAAgB,CAAC,OAAO,EAAE,OAAO,EAAE,WAAW,CAAC,CAAA;aAChD;QACH,CAAC,EAAE,SAAS,EAAC,yBAAyB,aAAgB,CAC3D,CAAA;AACH,CAAC,CAAA;AARY,QAAA,YAAY,gBAQxB"}