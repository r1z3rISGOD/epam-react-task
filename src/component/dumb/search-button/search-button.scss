@import "./src/_variables/variables.scss";

.search-settings__button {
  width: 275px;
  height: 50px;
  color: white;
  background-color: $color;
  border: none;
  border-radius: 3px;
  font-weight: bold;
  font-size: 1.5rem;
  font-family: $font;
  cursor: pointer;
}
