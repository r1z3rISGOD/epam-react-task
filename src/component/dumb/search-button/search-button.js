"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchButton = void 0;
var React = require("react");
require("./search-button.scss");
var SearchButton = function (_a) {
    var request = _a.request, setting = _a.setting, resultsSort = _a.resultsSort, fetchingAllFilms = _a.fetchingAllFilms;
    return (React.createElement("button", { onClick: function () {
            if (request !== '') {
                fetchingAllFilms(request, setting, resultsSort);
            }
        }, className: "search-settings__button" }, "SEARCH"));
};
exports.SearchButton = SearchButton;
//# sourceMappingURL=search-button.js.map