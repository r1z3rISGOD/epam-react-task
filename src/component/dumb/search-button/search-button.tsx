import * as React from 'react'
import './search-button.scss'
import { useDispatch } from 'react-redux'
import { changeSearchRequestedData, changeTypeOfSearch, putFetchedArrayToStore } from '../../../store/actions'
import { fetchData } from '../../services/fetchService'
import { useHistory } from 'react-router-dom'

interface SearchButtonProps {
  request: string
  setting: string
  resultsSort: string
}

export const SearchButton : React.FC<SearchButtonProps> = ({ request, setting, resultsSort }) => {
  const history = useHistory()
  const dispatch = useDispatch()

  return (
        <button onClick={async () => {
          if (request !== '') {
            const response = await fetchData(request, setting, resultsSort)
            dispatch(changeSearchRequestedData(request))
            dispatch(changeTypeOfSearch(setting))
            dispatch(putFetchedArrayToStore(response.data))
            history.push(`/search/?request=${request}`)
          }
        }} className="search-settings__button">SEARCH</button>
  )
}
