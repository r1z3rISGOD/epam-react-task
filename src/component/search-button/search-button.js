import React from "react";
import "./search-button.scss"

export default function SearchButton() {
    return(
        <button className="search-settings__button">SEARCH</button>
    )
}
