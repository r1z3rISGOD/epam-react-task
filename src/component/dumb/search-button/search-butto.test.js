import { SearchButton } from './search-button.tsx'
import React from 'react'

function f () {
  return 1
}

const setUp = (request) => shallow(<SearchButton request={request} setting='title' resultsSort='genres' fetchingAllFilms={f}/>)

describe('search button tests', () => {
  let component
  beforeEach(() => {
    component = setUp()
  })

  it('search button test 1', () => {
    const component = setUp('harry')
    expect(component).toMatchSnapshot()
    const btn = component.find('.search-settings__button')
    btn.simulate('click')
  })
  it('search button test 2', () => {
    const component = setUp('')
    expect(component).toMatchSnapshot()
    const btn = component.find('.search-settings__button')
    btn.simulate('click')
  })
})
