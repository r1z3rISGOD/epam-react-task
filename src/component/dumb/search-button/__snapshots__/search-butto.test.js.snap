// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search button tests search button test 1 1`] = `
<button
  className="search-settings__button"
  onClick={[Function]}
>
  SEARCH
</button>
`;

exports[`search button tests search button test 2 1`] = `
<button
  className="search-settings__button"
  onClick={[Function]}
>
  SEARCH
</button>
`;
